### PR TITLE
k8s For Mac and add Rancher Desktop

### PR DESCRIPTION
**Pull request by kevinbarbour**

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -1 +1,7 @@
-archive
\ No newline at end of file
+archive
```
Comment (mmatczuk) [inline .dockerignore:1]: There is no archive.

File: Makefile [hunk 2]
```diff
@@ -16,20 +16,30 @@ install-dependencies:
...
-k8s-setup-local: check-minikube check-skaffold
+.PHONY: minikube-setup
+minikube-setup:
 	minikube start --profile local-dev
-	skaffold config set --global local-cluster true
 	@eval $(minikube -p local-dev docker-env)
 
-k8s-local: check-skaffold
+.PHONY: rancher-setup
+rancher-setup:
```
Comment (mmatczuk) [inline Makefile:29]: I'd suggest setup-k8s-[rancher,minikube]

File: Makefile [hunk 3]
```diff
@@ -16,20 +16,30 @@ install-dependencies:
...
+	kubectl config use-context rancher-desktop
+
+.PHONY: skaffold
+skaffold: check-skaffold
+	skaffold config set --global local-cluster true
 	@skaffold dev
```
Comment (mmatczuk) [inline Makefile:35]: Let's remove the checks and add scaffold to install-dependencies with brew on Mac.
For inspiration see [here](https://github.com/mmatczuk/scylla-go-driver/blob/main/Makefile#L21).

File: k8s/docker/base.Dockerfile [hunk 4]
```diff
@@ -1,9 +1,15 @@
+FROM alpine:latest as builder
+WORKDIR /go/src/github.com/scylladb/scylla-cloud-poc
+RUN apk add --update make go --no-cache
+COPY . .
+RUN make build
+
```
Comment (mmatczuk) [inline k8s/docker/base.Dockerfile:6]: Nah, let's use local builds, it would make better use of cache plus on Mac the perf of file access is not there.

Comment (kevinbarbour) [reply k8s/docker/base.Dockerfile]: Changed so we cross-compile for linux locally on Mac as `scylla-cloud-linux` and symlink that to the binary on linux.

**Pull request merged**